eslint.config.js

```diff
@@ -52,7 +52,7 @@ export default tseslint.config(
     },
     settings: {
       node: {
-        version: '^18.0.0 || ^20.0.0 || >=22.0.0',
+        version: '^20.0.0 || >=22.0.0',
```

Comment:
`require(ESM)` is only supported (without a flag) for Node 20.19.0+, 22.12.0.
https://nodejs.org/docs/latest-v20.x/api/modules.html#loading-ecmascript-modules-using-require
Do we want to bump the version to that version? Or would it be fine to say if you want to use CJS, update the Node version?

Reply:
I think we should require Node 20.19.0+, 22.12.0+. To keep things cleaner, maybe we could merge this PR as "remove node 18 support" as it removes several branches we needed for it, and then propose the `require(esm)` by default range in another PR?

Reply:
Sounds good 👍